server/lib/adapters/mysql.js

```diff
@@ -112,24 +174,25 @@ export function runQuery(client, query, options = {}) {
  *
  * @returns {Promise} A promise object that wraps a stream.
  */
-export function streamQuery(client, query, options = {}) {
-  return new Promise((resolve, reject) => {
-    // Connect the connection.
-    client.connect((connectionError) => {
-      if (connectionError) {
-        connectionError.status = 401;
-        return reject(connectionError);
-      }
-
-      const params = { sql: query };
-
-      if (options.timeout && options.timeout > 0) {
-        params.timeout = options.timeout;
-      }
+function streamQuery(client, query, options = {}) {
+  return client.connect().then((conn) => {
+    const stream = conn.queryStream(query);
 
-      // Run the query.
-      return resolve(client.query(params).stream({ highWaterMark: 10 }));
+    stream.on("end", () => {
+      client.closeConnection();
```

Comment:
I'm going to need to look at this one closer, seems like it's a little different